spec/iex/endpoints/key_stat_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe IEX::Endpoints::KeyStat do
+  include_context 'client'
+
+  let(:stat) { 'ttmDividendRate' }
+  let(:symbol) { 'VTI' }
+
+  context 'when asking for a single stat', vcr: { cassette_name: 'key_stat/individual' } do
+    subject { client.key_stat(symbol, stat) }
+
+    it 'returns that stat in the type provided by the API' do
+      expect(subject).to be_within(0.0001).of(2.7887)
+    end
+  end
+
+  context 'invalid symbol', vcr: { cassette_name: 'key_stat/invalid_symbol' } do
+    subject { client.key_stat('INVALID', stat) }
+
+    it 'fails with SymbolNotFoundError' do
+      expect { subject }.to raise_error IEX::Errors::SymbolNotFoundError, 'Symbol INVALID Not Found'
+    end
+  end
+
+  context 'invalid stat', vcr: { cassette_name: 'key_stat/invalid_stat' } do
+    subject { client.key_stat(symbol, 'INVALID') }
+
+    it 'returns a hash of the API response' do
```

Comment:
This is a weird API. So if you specify something invalid it fails silently and returns the whole stats object? That's just bad design - I don't expect an API to succeed and return a different type of object. Shall we have the client raise a specialized exception in this case?

Reply:
I didn't either 🤷  I originally wrote the test to raise an error but was surprised when the test failed because the request "succeeded." I'm in favor of straight forward APIs w/ little nuance so an exception would be good here.